tests/phpunit/tests/query/cacheResults.php

```diff
@@ -67,28 +57,65 @@ public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
 		);
 	}
 
-	function set_up() {
-		parent::set_up();
-		$this->cache_args  = null;
-		$this->new_request = null;
-		add_filter( 'wp_query_cache_key', array( $this, 'filter_wp_query_cache_key' ), 15, 3 );
+
+	/**
+	 * @dataProvider data_query_cache
+	 * @covers WP_Query::generate_cache_key
+	 * @ticket 56802
+	 */
+	public function test_generate_cache_key( $args ) {
+		global $wpdb;
+		$query1 = new WP_Query();
+		$query1->query( $args );
+
+		$query_vars             = $query1->query_vars;
+		$request                = $query1->request;
+		$request_no_placeholder = $wpdb->remove_placeholder_escape( $request );
+
+		$reflection = new ReflectionMethod( $query1, 'generate_cache_key' );
+		$reflection->setAccessible( true );
+
+		$result_1 = $reflection->invoke( $query1, $query_vars, $request );
+
+		$result_2 = $reflection->invoke( $query1, $query_vars, $request_no_placeholder );
+
+		$this->assertSame( $result_1, $result_2 );
```

Comment:
Some additional assertions are needed:

- `$request ` contains the placeholder
- `$request_no_placeholder` does not

This will require a custom data provider for this test that only includes queries that contain the placeholder

Reply:
No, Peter, that doesn't work -- queries without the placeholder need to be tested too. This idea is "speak in the third person" bad.

There need to be tests for both queries with and without the placeholder. What I would like to do is ensure that at least one test runs in which the placeholder is initially present and replaced. Time to head to the docs.